1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,4 +1,48 @@
-function setAlarm() {}
+let flashParameter = 0;
+let flashIntervalBackground;
+
+function setAlarm() {
+  let timeOnScreen = document.getElementById("timeRemaining");
+  let inputTimeSeconds = document.getElementById("alarmSet");
+  let seconds = inputTimeSeconds.value;
+  inputTimeSeconds.value = "";
+  timeOnScreen.innerText = `Time Remaining: ${timeFormatter(seconds)}`;
+
+  let countDownInterval = setInterval(() => {
+    seconds--;
+    timeOnScreen.innerText = `Time Remaining: ${timeFormatter(seconds)}`;
+    if (timeOnScreen.innerText === "Time Remaining: 00:00") {
+      playAlarm();
+      clearInterval(countDownInterval);
+      flashInterval();
+    }
+  }, 1000);
+}
+
+function timeFormatter(sec) {
+  let minutes = Math.floor(sec / 60);
+  let seconds = sec % 60;
+  minutes = minutes < 10 ? "0" + minutes : minutes;
+  seconds = seconds < 10 ? "0" + seconds : seconds;
+  return `${minutes}:${seconds}`;
+}
```

Comment:
It's good that you have separated this code into its own function. It shows that you recognise sections of code that have a specific purpose and which have the potential to be reused.